2000-05-22  Jakub Jelinek  <jakub@redhat.com>

	* sysdeps/i386/fpu/bits/mathinline.h (__expm1_code): Avoid using ?:
	with omitted middle operand.

--- libc/sysdeps/i386/fpu/bits/mathinline.h	4 Jun 2003 22:17:35 -0000	1.1.1.13
+++ libc/sysdeps/i386/fpu/bits/mathinline.h	4 Jun 2003 22:22:36 -0000	1.11
@@ -381,7 +381,8 @@ __sincosl (long double __x, long double 
     ("fscale			# 2^int(x * log2(e))\n\t"		      \
      : "=t" (__temp) : "0" (1.0), "u" (__exponent));			      \
   __temp -= 1.0;							      \
-  return __temp + __value ?: __x
+  __temp += __value;							      \
+  return __temp ? __temp : __x
 __inline_mathcodeNP_ (long double, __expm1l, __x, __expm1_code)
 
 
